Add first gateways blog post (#422)
ipfs/ipfs-blog: src/_blog/break-on-through-gateways-part-1.md

![Content addressing](../assets/content-addressing.png)

The diagram illustrates what two different files look like on the network. The red jpeg represents one CID hosted on two nodes, while the purple jpeg represents a different CID hosted on two other nodes.

> **Note:** Depending on the size, IPFS may chunk (split) a single file into multiple blocks each with a CID of their own for efficiency reasons. Even so, the file will also have root CID. You can explore what this looks like for the NASA image using the [IPLD explorer](https://explore.ipld.io/#/explore/bafybeibml5uieyxa5tufngvg7fgwbkwvlsuntwbxgtskoqynbt7wlchmfm).

### You can ask any IPFS node for a CID

One of the benefits of content addressing is that you can retrieve a CID from any IPFS node as long as there's at least one node providing it to the network.

For example, if you ask an IPFS node for a CID that it doesn't have, the node can search the IPFS network, and if it finds another reachable IPFS node hosting the CID, it can fetch it and serve it back to you.

## Speaking IPFS

> It's worth noting that IPFS is a set of open-source protocols, specifications, and software implementations.

Reviewer: Maybe a bit distracting and not immediately relevant here, despite being an important topic project-wide right now.

So how do you use IPFS to access files in real-world applications?

> There are two prominent ways to fetch files stored in the IPFS network:

Reviewer: ```suggestion
There are two primary ways to fetch files stored in the IPFS network:
```

- Running an IPFS node by installing one of the IPFS implementations as a daemon (long-running process) on your computer or a server in the cloud. The node becomes a member of the IPFS peer-to-peer network and announces what data it’s holding and responds to requests for data.
- Using an **IPFS Gateway** which allows fetching CIDs using the HTTP protocol.

> The first option allows you to _speak the native IPFS protocol_ while the latter serves as a bridge in situations where you might be constrained to using HTTP. Choosing the right approach depends on your use case.

Reviewer: ```suggestion
The first option allows you to _speak the native IPFS protocol_. The latter serves as a bridge in situations where you might be constrained to using HTTP, such as in web applications where people visiting your site may not be running an IPFS node. Choosing the right approach depends on your use case.
```
...or something like it, to give more examples of the latter option

## What are IPFS gateways?

IPFS gateways are public services that translate between _Web2_ and _Web3_ thereby providing a bridge between HTTP and IPFS.

> They allow you to use the HTTP protocol –which almost every programming language is capable of– to request a CID from the IPFS network.

Reviewer: ```suggestion
They allow you to use the HTTP protocol – which almost every programming language is capable of – to request data by its CID from the IPFS network.
```

> You request content from an IPFS gateway by passing a CID in the HTTP request. Since CIDs are a hash of specific data, if the data is provided by a node on the network and accessible by the gateway, the gateway can get that specific data from the network no matter where it sits.

Reviewer: ```suggestion
You request data from an IPFS gateway by passing a CID in the HTTP request. Since CIDs are a hash of specific data, if the data is provided by a node on the network and accessible by the gateway, the gateway can get that specific data from the network no matter where it sits.
```
Keep it consistent to either "data" or "content", especially within the same paragraph.

Reviewer: Could add 2 things to make gateways even more concrete:
- A link to the Public Gateway checker: https://ipfs.github.io/public-gateway-checker/
- A pseudocode example of how to structure a gateway request, like `https://<gatewayurl>/<cid>`.

> In its simplest form, a gateway is an IPFS node that also accepts HTTP requests for CIDs in addition to speaking the IPFS protocol to participate in the peer-to-peer network. Most IPFS implementations can also work as a gateway.

Reviewer: ```suggestion
In its simplest form, a gateway is a computer running an IPFS node that also accepts HTTP requests for CIDs.
```
I think the second point about "most IPFS implementations" adds complexity that isn't as useful to readers learning about how to use (vs run) gateways.

> ![Any supported IPFS node can also be a gateway](../assets/browser-gateway.png)

Reviewer: ```suggestion
```
Related to suggestion for line 130, this (beautiful!) diagram could also be deleted.

### Example

To get a sense of using an IPFS gateway, try opening the following gateway links with the CID of the image of Astronaut Jessica Watkins:

- [https://ipfs.io/ipfs/bafybeibml5uieyxa5tufngvg7fgwbkwvlsuntwbxgtskoqynbt7wlchmfm](https://ipfs.io/ipfs/bafybeibml5uieyxa5tufngvg7fgwbkwvlsuntwbxgtskoqynbt7wlchmfm)
- [https://cloudflare-ipfs.com/ipfs/bafybeibml5uieyxa5tufngvg7fgwbkwvlsuntwbxgtskoqynbt7wlchmfm](https://cloudflare-ipfs.com/ipfs/bafybeibml5uieyxa5tufngvg7fgwbkwvlsuntwbxgtskoqynbt7wlchmfm)

As you might notice, these are two different gateways that are capable of fetching the same image using IPFS' core pillars: content addressing and peer-to-peer networking.

> This example only scratches the surface, leaving out many important details. The reality of using IPFS gateways is a nuanced topic that involves a space of trade-offs and details worth considering when developing applications.

Reviewer: Maybe more explicitly say where & how to learn about these nuances, or eliminate this paragraph.

## Summary

In this blog post, you learned about the challenges of the client-server model, the principles behind IPFS, namely peer-to-peer networking and content addressing, and how IPFS gateways provide a bridge between Web2 and Web3, allowing you to tap into the IPFS network using HTTP.

In the follow-up blog post, you will learn more about all the tips and tricks for using IPFS gateways in real-world applications, going into resolution styles, integration with DNS, caching, pinning, debugging, and more.

> If you're interested in:

Reviewer: love this way to wrap up
